### PR TITLE
Fix returned array of FiltersAggregation

### LINKED ISSUE #91: FiltersAggregation generates invalid array
The [FiltersAggregation](https://github.com/ongr-io/ElasticsearchDSL/blob/6ba9a7810e6aa5653de79dd2615016135ead408e/src/Aggregation/FiltersAggregation.php#L80-L82) adds filters inside an array with the type as key. Though, this generates the following array for a `TermQuery` with the 'term'-key twice in the stack:

``` php
array(
    'filtername' => array(
        'term' => array(
            'term' => array(
                'fieldname' => 'fieldvalue'
            )
        )
    )
)
```

This filter will throw an exception when sending it with a query to an Elasticsearch server.

I can supply a PR to fix this issue if you want.

### PR DESCRIPTION
**Pull request by niels-nijens**
* Added integration tests of documentation examples.
* Fixed unit tests of `FilterAggregation`.
* Fixed `FilterAggregation`.

Fixes #91.

### CODE REVIEW

**Comment by niels-nijens** [comment]
I was unsure where to put the integration tests. I'll be happy to move them somewhere else if you want. :smiley:

File: tests/Aggregation/FiltersAggregationTest.php [hunk 1]
```diff
@@ -11,14 +11,14 @@
...
 
-use ONGR\ElasticsearchDSL\Aggregation\AbstractAggregation;
 use ONGR\ElasticsearchDSL\Aggregation\FiltersAggregation;
 use ONGR\ElasticsearchDSL\BuilderInterface;
+use PHPUnit_Framework_TestCase;
 
 /**
  * Unit test for filters aggregation.
  */
-class FiltersAggregationTest extends \PHPUnit_Framework_TestCase
```
Comment (saimaz) [inline tests/Aggregation/FiltersAggregationTest.php]: To use `\` is completely fine here.

**Comment by saimaz** [comment]
In your particular case it's an integration test. So far we were using only unit tests which mirrored file structure in to the `test` folder. I'm thinking about two options here:
1. create Integration folder in tests folder and put that kind of tests there. Eventualy I'm thinking we will come up with more cases to test chains like you introduced.
2. put your test case in `SearchTest`.

@mvar, maybe you have any thoughts?

**Comment by mvar** [comment]
@saimaz I think it would be fine to simply move these tests into `SearchTest` like you have suggested.

**Comment by niels-nijens** [comment]
Alrighty. I will update the PR tomorrow.

**Comment by saimaz** [comment]
Thanks

**Comment by niels-nijens** [comment]
Rebased and updated the PR with the feedback.

**Pull request merged**